includes/admin/class-wc-rest-payments-orders-controller.php

```diff
@@ -139,19 +139,26 @@ public function capture_terminal_payment( $request ) {
 				$intent->get_currency()
 			);
 
-			// Capture the intent and update the order attributes.
-			$result = $this->gateway->capture_charge( $order );
-			if ( 'succeeded' !== $result['status'] ) {
-				$http_code = $result['http_code'] ?? 502;
-				return new WP_Error(
-					'wcpay_capture_error',
-					sprintf(
+			if ( 'succeeded' === $intent->get_status() ) {
```

Comment:
I'd suggest giving more context here:

```php
			// The delay until store receives event might be longer that Mobile app needed to complete order.
			$is_intent_succeeded         = 'succeeded' === $intent->get_status();
			$result_for_succeeded_intent = [
				'status' => 'succeeded',
				'id'     => $intent->get_id(),
			];
			// Capture the intent and update the order attributes.
			$result = $is_intent_succeeded ? $result_for_succeeded_intent : $this->gateway->capture_charge( $order );
			if ( 'succeeded' !== $result['status'] ) {
				$http_code = $result['http_code'] ?? 502;
				return new WP_Error(
					'wcpay_capture_error',
					sprintf(
					// translators: %s: the error message.
						__( 'Payment capture failed to complete with the following message: %s', 'woocommerce-payments' ),
						$result['message'] ?? __( 'Unknown error', 'woocommerce-payments' )
					),
					[ 'status' => $http_code ]
				);
			}
```

Reply:
Thanks for the tip. Addressed in d43ba1f.